core/lib/zksync_core/src/sync_layer/fetcher.rs

```diff
@@ -168,21 +168,23 @@ impl IoCursor {
 /// Errors that can be emitted by [`MainNodeFetcher`].
 #[derive(Debug, thiserror::Error)]
 enum FetcherError {
-    #[error("JSON-RPC error communicating with main node")]
+    #[error("JSON-RPC error communicating with main node: {0}")]
     Web3(#[from] EnrichedClientError),
-    #[error("internal fetcher error")]
+    #[error("internal fetcher error: {0:#}")]
     Internal(#[from] anyhow::Error),
 }
 
+pub(crate) fn is_transient(err: &EnrichedClientError) -> bool {
```

Comment:
Nit: I think at this point, it could make sense to move this logic to an `EnrichedClientError` method; it seems to be more-or-less general-purpose.

Reply:
done